### PR TITLE
Fix logic async gas consumption

### PR DESCRIPTION
**Pull request by amimart**
## Context

When prolog executing query through the logic `Ask` gRPC query, the prolog predicates are executed in a separate goroutine. In case this goroutine panic we are not able to recover from it, so the https://github.com/okp4/okp4d/pull/312 introduced a change in the gas consumption handling, making the goroutine stop instead of panic. This works on async parts, but may be problematic on synchronous parts.

## Description

Since https://github.com/ichiban/prolog/pull/290 released in the [v1.1.0 version](https://github.com/ichiban/prolog/releases/tag/v1.1.0) any panics occurring while executing predicates are recovered and an error is returned back to the interpreter client through the `Solutions.Err()` func. The panic being recovered, this PR adds a check on gas consumption to panic again with the `ErrorOutOfGas` error letting the gRPC `Ask` query handle it.

### CODE REVIEW

File: x/logic/keeper/interpreter.go [hunk 1]
```diff
@@ -43,6 +43,7 @@ func (k Keeper) execute(ctx goctx.Context, program, query string) (*types.QueryS
 	}
 
 	if err := i.ExecContext(ctx, program); err != nil {
+		// sdkCtx.GasMeter().ConsumeGas(sdkCtx.GasMeter().GasConsumed(), types.ModuleName)
```
Comment (ccamel) [inline x/logic/keeper/interpreter.go:46]: :stuck_out_tongue:

```suggestion
```

Comment (amimart) [reply x/logic/keeper/interpreter.go]: Oh, sorry for that 😿

**Review by ccamel — APPROVED** [review]
Super! 😌

**Review by bdeneux — APPROVED** [review]
Great 😍!

**Pull request merged**